File: EaterSpring/src/main/kotlin/xyz/purema/eater/controllers/ProductOrderController.kt
package xyz.purema.eater.controllers

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import xyz.purema.eater.dtos.ProductOrderDto
import xyz.purema.eater.dtos.query.ProductOrderQuery
import xyz.purema.eater.interfaces.IProductOrderService

@RestController
@RequestMapping("/v1/orders")
class ProductOrderController(
    private val productOrderService: IProductOrderService
) {
    @GetMapping
    fun getOrders(query: ProductOrderQuery): List<ProductOrderDto> = when {
        query.productId?.isNotEmpty() == true -> productOrderService.getByProductId(query.productId!!)
        query.customerId?.isNotEmpty() == true -> productOrderService.getByCustomerId(query.customerId!!)
        else -> productOrderService.getAllOrders()
    }

    @GetMapping("{id}")
    fun getOrderById(@PathVariable id: String): ProductOrderDto = productOrderService.getById(id)
}

File: EaterNet/Eater/Controllers/ProductOrderController.cs
using Eater.Dtos;
using Eater.Dtos.Query;
using Eater.Interfaces;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace Eater.Controllers
{
    [Route("v1/orders")]
    [ApiController]
    [Authorize]
    public class ProductOrderController : ControllerBase
    {
        private readonly IProductOrderService _productOrderService;

        public ProductOrderController(IProductOrderService productOrderService)
        {
            _productOrderService = productOrderService;
        }

        [HttpGet]
        public async Task<List<ProductOrderDto>> GetOrders([FromQuery] ProductOrderQuery query)
        {
            List<ProductOrderDto> orders;

            if (!string.IsNullOrEmpty(query.ProductId))
            {
                orders = await _productOrderService.GetAllByProductId(query.ProductId);
            }
            else if (!string.IsNullOrEmpty(query.CustomerId))
            {
                orders = await _productOrderService.GetAllByCustomerId(query.CustomerId);
            }
            else
            {
                orders = await _productOrderService.GetAll();
            }

            return orders;
        }

        [HttpGet("{id}")]
        public async Task<ProductOrderDto> GetOrderById(string id) => await _productOrderService.GetById(id);
    }
}
